cmd/secret-collection-manager/main.go

```diff
@@ -0,0 +1,419 @@
+	user, err := m.userByAliasCached(userName)
+	if err != nil {
+		return fmt.Errorf("failed to get user %s: %w", userName, err)
+	}
+	policy := managedVaultPolicy{Path: map[string]managedVaultPolicyCapabiltyList{
+		m.kvMetadataPrefix + "/" + secretCollectionName + "/*": {Capabilities: []string{"list"}},
+		m.kvDataPrefix + "/" + secretCollectionName + "/*":     {Capabilities: []string{"create", "update", "read"}},
+	}}
+	serializedPolicy, err := json.Marshal(policy)
+	if err != nil {
+		return fmt.Errorf("failed to serialize policy for %s: %w", secretCollectionName, err)
+	}
+	if err := m.privilegedVaultClient.Sys().PutPolicy(prefixedName(secretCollectionName), string(serializedPolicy)); err != nil {
+		println(string(serializedPolicy))
+		return fmt.Errorf("failed to create policy %s: %w", prefixedName(secretCollectionName), err)
+	}
+
+	group := vaultclient.Group{
+		Name:            prefixedName(secretCollectionName),
+		Policies:        []string{prefixedName(secretCollectionName)},
+		MemberEntityIDs: []string{user.ID},
+		Metadata:        map[string]string{"created-by-secret-collection-manager": "true"},
+	}
+	serializedGroup, err := json.Marshal(group)
+	if err != nil {
+		return fmt.Errorf("failed to marhsal group: %w", err)
+	}
+	if err := m.privilegedVaultClient.Put("identity/group", serializedGroup); err != nil {
+		return fmt.Errorf("failed to create group %s: %w", prefixedName(secretCollectionName), err)
+	}
+
+	return nil
+}
+
+func prefixedName(name string) string {
+	return objectPrefix + "-" + name
+}
+
+func (m *secretCollectionManager) listSecretCollections(l *logrus.Entry, user string, w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	collections, err := m.getCollectionsForUser(l, user)
+	if err != nil {
+		l.WithError(err).Error("failed to get collections")
+		http.Error(w, fmt.Sprintf("failed to get secret collections. RequestID: %s", l.Data["UID"]), 501)
+		return
+	}
+
+	if len(collections) == 0 {
+		w.WriteHeader(http.StatusOK)
+		return
+	}
+	serialized, err := json.Marshal(collections)
+	if err != nil {
+		l.WithError(err).Error("failed to serialize")
+		http.Error(w, fmt.Sprintf("failed to serialize. RequestID: %s", l.Data["UID"]), 501)
+		return
+	}
+
+	if _, err := w.Write(serialized); err != nil {
+		l.WithError(err).Error("failed to write response")
+	}
+}
+
+func (m *secretCollectionManager) getCollectionsForUser(_ *logrus.Entry, userName string) ([]secretCollection, error) {
+	user, err := m.userByAliasCached(userName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get user %s: %w", userName, err)
+	}
+
+	var groupNames []string
+	for _, groupID := range user.GroupIDs {
+		name, err := m.groupNameFromId(groupID)
+		if err != nil {
+			return nil, fmt.Errorf("failed to resolve name for group id %s: %w", groupID, err)
+		}
+		if !strings.HasPrefix(name, objectPrefix) {
+			continue
+		}
+		groupNames = append(groupNames, name)
```

Comment:
I'd consider `if HasPrefix(...) { groupNames = append(...) }` a bit more readable (getting rid of the negation and `continue`) but /shrug